packages/core/chrome/core-chrome-browser-internal/src/project_navigation/project_navigation_service.ts

```diff
@@ -88,4 +108,48 @@ export class ProjectNavigationService {
       },
     };
   }
+
+  private setActiveProjectNavigationNodes(_location?: Location) {
+    if (!this.application) return;
+    if (!Object.keys(this.projectNavigationNavTreeFlattened).length) return;
+
+    const location = _location ?? this.application.history.location;
+    let currentPathname = this.http?.basePath.prepend(location.pathname) ?? location.pathname;
+
+    // We add possible hash to the current pathname
+    // e.g. /app/kibana#/management
+    currentPathname = stripQueryParams(`${currentPathname}${location.hash}`);
+
+    const activeNodes = findActiveNodes(
+      currentPathname,
+      this.projectNavigationNavTreeFlattened,
+      location
+    );
+
+    let requiresUpdate = false;
+
+    if (activeNodes.length !== this.activeNodes$.value.length) {
+      requiresUpdate = true;
+    } else {
+      this.activeNodes$.value.forEach((nodesBranch, i) => {
+        nodesBranch.forEach((node, j) => {
+          if (node.id !== activeNodes[i][j]?.id) {
+            requiresUpdate = true;
```

Comment:
nit: I think this should be two `for` loops with a `break`

Reply:
I thought about it and then decided that the tiny list (less than 100) would not benefit of the optimisation.

Reply:
I also think a `for` loop is more readable than a nested forEach in this case. `for` loop with i/j is at a glance associated with looping over a matrix, but it took longer for me to parse the nested forEach

Reply:
Changed in https://github.com/elastic/kibana/pull/159906/commits/bd7848ddafe8ebf4c98efe3231bc305a7e48b606